internal/faustjs.org/docs/guides/custom-post-types.mdx

```diff
@@ -0,0 +1,335 @@
+
+<img
+  src="/docs/img/cpt-guide/movie-single-template.png"
+  alt="The finished Faust template for the single movie being resolved in the browser"
+/>
+
+### Archive Movie Template
+
+With our single movie template created, let's create our template for the movie archive next.
+
+If you navigate to [`http://localhost:3000/movies`](http://localhost:3000/movies) you will get a 404. Inspect the server output, and you will see the possible templates for the route:
+
+<img
+  src="/docs/img/cpt-guide/possible-templates-movie-archive.png"
+  alt="A screenshot of the output from Faust describing the possible templates for the route"
+/>
+
+You can see using any of the following templates will resolve for this route:
+
+```js
+['archive-movies', 'archive', 'index'];
+```
+
+Since we want this template to be just for our movie archive, we'll create a template called `archive-movies`. In your `wp-templates` directory, create a file called `archive-movies.js` with the following content:
+
+```js
+import { gql } from '@apollo/client';
+import Link from 'next/link';
+
+export default function ArchiveMovies(props) {
+  const { label, contentNodes } = props.data.nodeByUri;
+
+  return (
+    <>
+      <h1>{label} Archive</h1>
+
+      <ul>
+        {contentNodes.nodes.map((node) => (
+          <li>
+            <Link href={node.uri}>{node.title}</Link>
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+}
+
+ArchiveMovies.variables = ({ uri }) => {
+  return { uri };
+};
+
+ArchiveMovies.query = gql`
+  query MovieArchive($uri: String!) {
+    nodeByUri(uri: $uri) {
+      ... on ContentType {
+        label
+        description
+        contentNodes {
+          nodes {
+            databaseId
+            uri
+            ... on NodeWithTitle {
+              title
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+```
+
+Don't forget to register your new archive template in the `wp-templates/index.js` file:
+
+```js title="wp-templates/index.js" {16}
+import category from './category';
+import tag from './tag';
+import frontPage from './front-page';
+import page from './page';
+import single from './single';
+import SingleMovie from './single-movies';
+import ArchiveMovies from './archive-movies';
+
+export default {
+  category,
+  tag,
+  'front-page': frontPage,
+  page,
+  single,
+  'single-movies': SingleMovie,
+  'archive-movies': ArchiveMovies,
+};
+```
+
+Our archive movie template has now been registered! If you visit [`http://localhost:3000/movies`](http://localhost:3000/movies) again you can see the template properly resolves:
+
+<img
+  src="/docs/img/cpt-guide/movie-archive-template.png"
+  alt="The finished Faust template for the movie archive being resolved in the browser"
+/>
+
+You now have two new Faust templates resolving for your Movies custom post type!
```

Comment:
Is there an easy way to link to the whole code? I've found people respond to that as a "done" version at the end of many of my tutorials.

Reply:
We would have to create an example project if we wanted to link them to a live code repo. However, a CPT example may not be worth it since the template names and fields are reliant on the data created in WordPress.